pmd-scala/src/test/java/net/sourceforge/pmd/scala/rule/XPathRuleTest.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.scala.rule;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.StringReader;
+
+import org.apache.commons.io.IOUtils;
+import org.junit.Before;
+import org.junit.Test;
+
+import net.sourceforge.pmd.PMD;
+import net.sourceforge.pmd.PMDException;
+import net.sourceforge.pmd.Report;
+import net.sourceforge.pmd.Rule;
+import net.sourceforge.pmd.RuleContext;
+import net.sourceforge.pmd.RuleSet;
+import net.sourceforge.pmd.RuleSetFactory;
+import net.sourceforge.pmd.RuleSets;
+import net.sourceforge.pmd.RuleViolation;
+import net.sourceforge.pmd.lang.LanguageRegistry;
+import net.sourceforge.pmd.lang.rule.XPathRule;
+import net.sourceforge.pmd.lang.rule.xpath.XPathRuleQuery;
+import net.sourceforge.pmd.lang.scala.ScalaLanguageModule;
+import net.sourceforge.pmd.testframework.RuleTst;
+
+public class XPathRuleTest extends RuleTst {
+    private static final String SCALA_TEST = "/parserFiles/helloworld.scala";
+
+    XPathRule rule;
+
+    @Before
+    public void setUp() {
+        rule = new XPathRule();
+        rule.setLanguage(LanguageRegistry.getLanguage(ScalaLanguageModule.NAME));
+        rule.setMessage("XPath Rule Failed");
+    }
+
+    @Test
+    public void testPrintHelloWorld() throws Exception {
+        String xpath = "//ASTTermApplyNode/ASTTermNameNode[@Image=\"println\"]";
```

Comment:
Yes, I'd say we should go without "AST" and without "Node"... Then this becomes: `//TermApply/TermName[@Image='println']`

Reply:
agreed.